arithmetic_analysis/lu_decomposition.py

```diff
@@ -6,7 +6,7 @@
 from __future__ import annotations
 
 import numpy as np
-import numpy.typing as NDArray
+import numpy.typing as NDArray  # noqa: N812
```

Comment:
```suggestion
from numpy.typing import ArrayLike
```
And then on lines 14 and 15 change `NDArray` --> `ArrayLike`